code/datums/wires/explosive.dm

```diff
@@ -11,6 +11,45 @@
 /datum/wires/explosive/proc/explode()
 	return
 
+/datum/wires/explosive/chem_grenade
+	holder_type = /obj/item/grenade/chem_grenade
+	randomize = TRUE
+	var/fingerprint
+	var/assembly
+
+/datum/wires/explosive/chem_grenade/interactable(mob/user)
+	var/obj/item/grenade/chem_grenade/G = holder
+	if(G.stage == 2)
+		return TRUE
+
+/datum/wires/explosive/chem_grenade/attach_assembly(color, obj/item/assembly/S)
+	if(istype(S,/obj/item/assembly/timer))
+		var/obj/item/grenade/chem_grenade/G = holder
+		var/obj/item/assembly/timer/T = S
+		G.det_time = T.saved_time*10
+	if(istype(S,/obj/item/assembly/prox_sensor))

```

Comment:
```suggestion
	else if(istype(S,/obj/item/assembly/prox_sensor))
```

If we know it's a timer we shouldn't even bother checking if it's a prox_sensor since it can't be both.